proto/intertx/tx.proto

```diff
@@ -24,17 +26,29 @@ message MsgRegisterAccount {
   // the channel by which the packet will be sent
   string source_channel = 3 [(gogoproto.moretags) = "yaml:\"source_channel\""];
 
-  // Timeout height relative to the current block height.
-  // The timeout is disabled when set to 0.
-  ibc.core.client.v1.Height timeout_height = 4
-      [(gogoproto.moretags) = "yaml:\"timeout_height\"", (gogoproto.nullable) = false];
-
-  // Timeout timestamp (in nanoseconds) relative to the current block timestamp.
-  // The timeout is disabled when set to 0.
-  uint64 timeout_timestamp = 5 [(gogoproto.moretags) = "yaml:\"timeout_timestamp\""];
   string owner=6;
 }
 
 message MsgRegisterAccountResponse {}
 
+// MsgSend represents a message to send coins from one account to another.
+message MsgSend {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // the port by which the packet will be sent
+  string chain_type = 1 [(gogoproto.moretags) = "yaml:\"chain_type\""];
+  string source_port = 2 [(gogoproto.moretags) = "yaml:\"source_port\""];
+
+  // the channel by which the packet will be sent
+  string source_channel = 3 [(gogoproto.moretags) = "yaml:\"source_channel\""];
+
+  bytes   sender                    = 6 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+  bytes   to_address                      = 7 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
```

Comment:
if interchain accounts is to ever interact with a non SDK based interchain acc implementation (like maybe in rust), these would need to be `string`, ibc-transfers uses strings for sender and receiving address as well

Reply:
Disregard the above. 

Why doesn't the send message just take in bytes of a preconstructed SDK bank message? This would follow the flow we discussed where the side sending the message doesn't know what type of message the packet contains, but just verifies the sender and relays that to the receiving side. The receiving side then decodes the message (for SDK it would decode into sdk.Msg) and check that the signer matches the authenticated sender

Reply:
Unless I am mistaken that is what we are doing, but we are building the message in the keeper here -> https://github.com/cosmos/interchain-account/blob/feat/send-inter-tx/x/inter-tx/keeper/relay.go#L21

Which gets called here -> https://github.com/cosmos/interchain-account/blob/feat/send-inter-tx/x/inter-tx/keeper/msg_server.go#L53

Reply:
ahh yes, I understand now. `inter-tx` is a helper module sending a `sdk.Msg` via the `ibc-account` module

I think the above construction should be good then